.gitlab-ci.yml

```diff
@@ -1,34 +1,43 @@
-image: docker:19.03.12stages:
+image: docker:19.03.12
+
+stages:
   - testing
   - release
 
 variables:
+  GITLABCI_VERSION:    "2.1"
   GITLABCI_IMAGE_TEST: $CI_REGISTRY_IMAGE:$CI_COMMIT_SHA
-  GITLABCI_IMAGE:      $CI_REGISTRY_IMAGE:2.1
+  GITLABCI_IMAGE:      $CI_REGISTRY_IMAGE:$GITLABCI_VERSION
 
 check-pr:
-  stage: testing 
+  stage: testing
+  services:
+    - docker:19.03.12-dind
   except:
     - master
-  #only:
-  #  changes:
-  #    - docker-gitlabci/*
+  only:
+    changes:
+      - docker-gitlabci/*
   script:
-    - ./docker-gitlabci/build.sh
+    - cd docker-gitlabci
+    - ./build.sh $GITLABCI_VERSION
 
 build-gitlabci:
+  environment: maintainer-approved
   stage: testing
   only:
     refs:
       - master
     changes:
       - docker-gitlabci/*
   script:
-    - ./docker-gitlabci/build.sh
+    - cd docker-gitlabci
+    - ./build.sh $GITLABCI_VERSION
     - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
     - docker push $GITLABCI_IMAGE_TEST
 
 image-gitlabci:
```

Comment:
Do we want another stage if we only do it on master anyway? Why not push directly to the correct image?

Reply:
Thats a good question...

I think its to meta to use the tests from domjudge:.gitlab-ci.yml here to test if you want this updated image, so I'll just remove the 2nd stage.